overview/helloWorld/4-iexecApplication.md

```diff
@@ -0,0 +1,247 @@
+# Build Your iDapp
+
+This guide will help you set up, test, run, and deploy your iDapp using the
+`idapp-cli` tool. Follow the instructions carefully for a smooth development
+experience.
+
+## Prerequisites
+
+Before getting started, ensure that you have the following installed on your
+system:
+
+- [**Node.js**](https://nodejs.org/en/) version 18 or higher
+- [**NPM**](https://docs.npmjs.com/) (Node.js package manager)
+- [**DockerHub**](https://hub.docker.com/) account
+
+## Installation
+
+First, you need to install the `idapp-cli` package. Open your terminal and run
+the following command:
+
+```sh
+npm install @iexec/idapp-cli
+```
+
+## Initialize Framework
+
+To initialize the working directory for developing your iDapp, use the
+`idapp init` command. This command sets up the necessary project structure and
+files.
+
+```sh
+idapp init
+```
+
+You will be prompted with the following message:
+
+```sh
+ ___ ____    _    ____  ____
+ |_ _|  _ \  / \  |  _ \|  _ \
+  | || | | |/ _ \ | |_) | |_) |
+  | || |_| / ___ \|  __/|  __/
+ |___|____/_/   \_\_|   |_|
+
+? A new project will be created in the current directory. Do you want to continue? Yes
+? What is the name of your project? develop-my-idapp
+? Which language do you want to use to build your iDapp? JavaScript
+? Would you like to access a protected data inside your iDapp? (Y/n)
+```
+
+:::tip
+
+Answer `n` to the question
+`? Would you like to access protected data inside your iDapp?`. We will cover
+integrating protected data into your iDapp later.
+
+:::
+
+## Test Your iDapp
+
+To test your iDapp, use the `idapp test` command. This command runs your iDapp
+to ensure that your application is functioning correctly.
+
+### Using Parameters
+
+You can pass input parameters to your iDapp using the `--param` option. This
+allows you to provide necessary inputs during runtime.
+
+```sh
+idapp test --param <your-input>
+```
+
+### Using Docker
+
+For a more realistic environment, you can run your iDapp using Docker. This
+helps in creating an environment that closely mimics production settings.
+
+```sh
+idapp run --docker
+```
+
+## Deploy Your iDapp
+
+Deploy your iDapp on the iExec protocol.
+
+```sh
+idapp deploy
+```
+
```

Comment:
I expect to find here: "If everything goes well, the result should look like: ..." and give the console output? idapp-address, sconified docker image, etc.